docs/api/grpc/overview.md

```diff
@@ -1,16 +1,152 @@
 ---
-title: gRPC API overview
+title: Using gRPC
+sidebar_label: Using gRPC
 hide_title: false
+description: Learn how to use the gPRC API.
+vega_network: TESTNET
 ---
-gRPC is a low-latency, highly efficient API for interacting with the network and querying for and filtering data.
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
 
-## Core node APIs
-Core nodes are responsible for ensuring the consensus rules are met and that a consistent view of the network is seen. They present endpoints that give access to the state of the network (block time, block height etc), allow transactions to be submitted to the network and to subscribe to event streams so that changes of internal state can be seen.
+gRPC is a low-latency, highly efficient communication method that allows a client and a server to send data over HTTP. Vega supports a gRPC API for interacting with the network and querying for and filtering data. 
 
-- [Core](vega/api/v1/core.proto.mdx): Submit transactions to the network and retrieve basic information about the node and network such as the block height, app version, number of peers, etc.
-- [Core state](vega/api/v1/corestate.proto): Request information about the present ‘state’ of the Vega node. Importantly, the core node only maintains enough state information such that it can process incoming transactions; it keeps no history beyond what is required for that function. You can request current lists of accounts, parties, proposals etc., which are held in memory by the core node.
+## gRPC API and Protocol Buffers
+gRPC uses Protocol Buffers (protobufs) as its underlying data definition. Protobuf messages are language-neutral data definitions that are used to generate language-specific data structures. Code generation of a protobuf message also generates functions to serialise each message so that they can be sent through gRPC to each RPC service endpoint.
 
-## Data node API
+This means that Vega's gRPC API is fully defined by its protobuf definitions, and so the protobuf definitions themselves provide complete documentation.
+
+## Relationship to REST API
+The buf ecosystem contains plugins that also allow automatic generation of REST API endpoints from the protobuf definitions. The result is that Vega's gRPC and REST API match exactly in both structure and functionality.
+
+As REST is a more familiar way to interact with a product than gRPC, experiencing Vega first through the REST API and using the REST documentation may help ease an initial integration, without any loss of functionality. Migrating later to using gRPC is then trivial since the mapping between REST and gRPC is 1-to-1.
```

Comment:
One thing unclear to me (mostly because I don't know how the gRPC plugins work) is how the REST  endpoints are generated from the protobuf definitions - I don't think I would be able to make the connection between specific gRPC messages and REST endpoints without the REST docs.

Is this saying I should find it trivial to work-out what a REST endpoint is just from the proto definitions?

Reply:
I think the idea is that most people would go the other way around -- they'll play with the REST API first because its more familiar, and then work out how to go from the REST docs to gRPC. So in that situation they'll be looking at, say, the docs for this REST endpoint:
https://docs.vega.xyz/mainnet/api/rest/data-v2/trading-data-service-get-order
which has the heading "Get Order", wonder how to do it in gRPC and start tapping `.GetO` into their IDE which would autocomplete into `.GetOrder()` and they go "wow that was easy, I don't even _need_ to look at the gRPC docs"

If you were to want to go the other way and go from a gRPC call to the REST URL for it, the true answer is that the map is in this file which we could consider:
https://github.com/vegaprotocol/vega/blob/develop/protos/sources/data-node/grpc-rest-bindings.yml
but you could also just look through in the REST API docs to find it and the way they are grouped in the sidebar should help.

@candida-d do you think we should add a link to https://github.com/vegaprotocol/vega/blob/develop/protos/sources/data-node/grpc-rest-bindings.yml ? I'm worried it'll just make it seem more confusing.

Reply:
I've added two snippets showing how similar the using REST and gRPC are (and linked to the yml file that maps the url <-> gRPC function call), which is hopefully is more useful